sunkit_instruments/response/thermal.py

```diff
@@ -0,0 +1,160 @@
+    spectra: `~astropy.units.Quantity`
+        Source spectra as a 2D array. The first axis should correspond to temperature and the
+        second axis should correspond to wavelength.
+    density: `~astropy.units.Quantity`, optional
+        1D array describing the variation in density along the temperature axis. It is assumed
+        that temperature and density are dependent.
+    meta: `dict`, optional
+        Any optional metadata to attach to the spectra, e.g. abundance model, CHIANTI version
+    """
+
+    @u.quantity_input
+    def __init__(
+        self,
+        temperature: u.K,
+        wavelength: u.Angstrom,
+        spectra: u.photon * u.cm**3 / (u.s * u.Angstrom * u.steradian),
+        density: u.cm ** (-3) = None,
+        meta=None,
+    ):
+        self.meta = meta
+        coords = {
+            "temperature": xarray.Variable(
+                "temperature",
+                temperature.value,
+                attrs={"unit": temperature.unit.to_string()},
+            ),
+            "wavelength": xarray.Variable(
+                "wavelength",
+                wavelength.value,
+                attrs={"unit": wavelength.unit.to_string()},
+            ),
+        }
+        if density is not None:
+            coords["density"] = xarray.Variable(
+                "temperature", density.value, attrs={"unit": density.unit.to_string()}
+            )
+        self._da = xarray.DataArray(
+            spectra.data,
+            dims=["temperature", "wavelength"],
+            coords=coords,
+            attrs={"unit": spectra.unit.to_string(), **self.meta},
+        )
+
+    def __repr__(self):
+        return self._da.__repr__()
+
+    def __str__(self):
+        return self._da.__str__()
+
+    def _repr_html_(self):
+        return self._da._repr_html_()
+
+    @property
+    def meta(self):
+        return self._meta
+
+    @meta.setter
+    def meta(self, x):
+        if x is None:
+            self._meta = {}
+        elif isinstance(x, dict):
+            self._mata = x
+        else:
+            raise TypeError(f'Unsupported metadata type {type(x)}')
+
+    @property
+    @u.quantity_input
+    def temperature(self) -> u.K:
+        return u.Quantity(self._da.temperature.data, self._da.temperature.attrs["unit"])
+
+    @property
+    @u.quantity_input
+    def wavelength(self) -> u.Angstrom:
+        return u.Quantity(self._da.wavelength.data, self._da.wavelength.attrs["unit"])
+
+    @property
+    @u.quantity_input
+    def data(self) -> u.photon * u.cm**3 / (u.s * u.Angstrom * u.steradian):
+        return u.Quantity(self._da.data, self._da.attrs["unit"])
+
+    @u.quantity_input
+    def temperature_response(
+        self, channel, obstime=None
+    ) -> u.cm**5 * u.DN / (u.pixel * u.s):
+        """
+        Temperature response function for a given instrument channel.
+
+        The temperature response function describes the sensitivity of an imaging
```

Comment:
That's a great question. I would hope that yes it could be extended to non-imaging instruments as well. What would the units be in that case? Also, do you have an example of a non-imaging instrument that computes a temperature response? Is it common to compute temperature responses for GOES XRS?